battery_manager.py

```diff
@@ -81,10 +82,11 @@ def balance(self) -> None:
                   'The system will not perform balancing')
             return
 
-        highest_cells = self.battery_system.highest_voltage_cells(4)
+        required_voltage: float = max(lowest_voltage + self.MIN_CELL_DIFF_FOR_BALANCING, SocCurve.soc_to_voltage(0.15))
```

Comment:
Behaviour Looks fine to me. I suggest passing this responsibility to the BatteryCell class to come up with the 15% soc voltage. I.e. create a soc_to_voltage method in BatteryCell. The benefit is that the BatteryManager class does not need to have knowledge which SocCurve the Battery Cell is using. Also we reduce dependencies by doing this. If we decide to introduce different kinds of BatteryCells, we do not need to change the balancing algorithm.